src/main/kotlin/com/hubelias/parkingmeter/port/adapter/rest/ParkedVehiclesEndpoint.kt

```diff
@@ -0,0 +1,49 @@
+package com.hubelias.parkingmeter.port.adapter.rest
+
+import com.hubelias.parkingmeter.application.ParkingMeterFacade
+import org.springframework.hateoas.mvc.ControllerLinkBuilder.linkTo
+import org.springframework.http.HttpHeaders
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.stereotype.Controller
+import org.springframework.web.bind.annotation.*
+import java.security.Principal
+import javax.annotation.security.RolesAllowed
+
+
+@Controller
+@RequestMapping("api/parkedVehicles")
+class ParkedVehiclesEndpoint(
+        private val parkingMeterFacade: ParkingMeterFacade
+) {
+    data class StartParkingCmd(val vehicleId: String)
+
+    @PostMapping
+    @RolesAllowed(UserRole.DRIVER)
+    fun startParking(@RequestBody cmd: StartParkingCmd, principal: Principal): ResponseEntity<Void> {
+        parkingMeterFacade.startParking(principal.name, cmd.vehicleId)
+
+        val headers = HttpHeaders().apply {
+            location = linkTo(ParkedVehiclesEndpoint::class.java).slash(cmd.vehicleId).toUri()
+        }
+
+        return ResponseEntity(headers, HttpStatus.CREATED)
```

Comment:
wydaje mi się że spring ma na to factory `created(...)`

Reply:
Ma. Nawet od niego zacząłem, ale potem chciałem jeszcze dodać HATEOASowe body. Ostatecznie zrezygnowałem z body, ale konstruktor zamiast factory został. `ResponseEntity.created` byłoby tu zdecydowanie bardziej eleganckie.